docs/source/essentials/data.md

```diff
@@ -2,108 +2,67 @@
 title: Fetching data
 ---
 
-GraphQL is the best way to work with data from **any** back-end that your product needs. It is not a mapping of your database, but rather a graph of the data sources and shapes your product is made of. Works with multiple data sources (first party or third)
+## Overview
 
-## Context
-
-> TODO: Shorten this up, a lot.
+GraphQL is the best way to work with data from **any** back-end that your product needs. It is not a mapping of your database, but rather a graph of the data sources and shapes your product is made of. Resolvers are the key to this graph. Each resolver represents a single field, and can be used to fetch data from any source(s) you may have.
 
-### What is?
+## Context
 
-The `context` is the third positional argument passed to every resolver. `context` references the same object across all resolvers, so no resolver should modify the contents. Additionally, it is best to ensure that the contents of `context` does not change depending on the particular query or mutation.
+The context is how you access your shared connections and fetchers in resolvers to get data.
 
-The common uses of the `context` are storing [authentication scope](), database connections([mongo](), [postgres](), etc), and custom fetch functions. Additionally when batching requests across different resolvers to avoid the n+1 problem, you will attach your instances of [dataloader](best-practice) to the `context`.
+The `context` is the third argument passed to every resolver. It is useful for passing things that any resolver may need, like [authentication scope](), database connections([mongo](), [postgres](), etc), and custom fetch functions. Additionally, if you're using [dataloaders to batch requests](../best-practices/performance.html#Batching-data-lookups)  across resolvers, you can attach them to the `context` as well.
 
+As a best practice, `context` should be the same for all resolvers, no matter the particular query or mutation, and resolvers should never modify it. This ensures consistency across resolvers, and helps increase development velocity.
 
 > (Evans) not sure if this idea of a constant context is completely true/a best-practice, expecially if making a connection is costly, so you only start the operation if certain fields are requested
```

Comment:
What should we do about this?

Reply:
@JakeDawkins I think we can remove this comment.

Making the database connection might be costly, but I think it would make sense to keep a handle to the database client (with its connection state potentially uncertain or disconnected) on the `context` either way.

Of course, such a database handle would only be initiated if a connection to the data source had been necessary.  If no query had been made, presumably the database client code would not have made the actual connection (this doesn't seem like our concern either way).  Furthermore, it would likely either be forcefully disconnected as an idle client by the server if it had gone unused, or it would voluntarily disconnect on its own accord after a certain amount of time.  When being used again, the client would take responsibility for the re-connection.  This incurs some rebuild time for some connections, but would yield a win for the majority (assuming queries weren't few and far between).